src/store/document-drive.ts

```diff
@@ -43,74 +48,237 @@ function debounceOperations(
+            (currentValue?.document &&
+                newValue?.document &&
+                documentToHash(currentValue.document) !==
+                    documentToHash(newValue.document))
+        ) {
+            // if document has been fetched then updates the cache
+            if (newValue?.status === 'LOADED') {
+                documentCache.set(
+                    `${newValue.driveId}:${newValue.documentId}`,
+                    newValue.document,
+                );
+            }
+
+            set(singletonFileNodeDocumentAtom, newValue);
+            return true;
+        }
+
+        return false;
+    },
+);
+
+const selectedDocumentAtom = selectAtom(
+    fileNodeDocumentAtom,
+    fileNode => fileNode?.document,
+);
+
 export function useFileNodeDocument(
-    props: TUiNodesContext & TDocumentDriveServer,
+    props: TUiNodesContext & TDocumentDriveServer & IReadModeContext,
 ) {
-    const { selectedNode, openFile, addOperations, onStrandUpdate } = props;
-    const [selectedDocument, setSelectedDocument] = useState<
-        Document | undefined
-    >();
+    const {
+        selectedNode,
+        selectedDriveNode,
+        openFile,
+        addOperations,
+        onStrandUpdate,
+        fetchDocument: fetchReadDocument,
+    } = props;
+    const [fileNodeDocument, setFileNodeDocument] =
+        useAtom(fileNodeDocumentAtom);
+    const isReadMode = selectedDriveNode?.syncStatus === undefined;
+    const driveId = selectedNode?.driveId;
+    const documentId = selectedNode?.id;
+    const kind = selectedNode?.kind;
+    const documentType =
+        kind === 'FILE' ? selectedNode?.documentType : undefined;
 
-    const fetchDocument = useCallback(async () => {
-        if (selectedNode?.kind !== FILE) {
-            return;
-        }
-        try {
-            const document = await openFile(
-                selectedNode.driveId,
-                selectedNode.id,
-            );
-            setSelectedDocument(document);
-        } catch (error) {
-            setSelectedDocument(undefined);
-            console.error(error);
+    const selectedDocument = useAtomValue(selectedDocumentAtom);
+    const setSelectedDocument = useMemo(() => {
+        function setSelectedDocument(document: Document | undefined) {
+            if (!document) {
+                setFileNodeDocument(undefined);
+            } else if (fileNodeDocument) {
+                setFileNodeDocument({ ...fileNodeDocument, document });
+            } else {
+                throw new Error('fileNodeDocument is undefined');
+            }
         }
-    }, [openFile, selectedNode?.driveId, selectedNode?.id, selectedNode?.kind]);
+        return setSelectedDocument;
+    }, [fileNodeDocument, setFileNodeDocument]);
+
+    const fetchDocument = useCallback(
+        async (driveId: string, id: string, documentType: string) => {
+            try {
+                const document = await (isReadMode
```

Comment:
The users of this hook just want to get the selected document, this calls the correct method depending if it belongs to a read drive or a synced drive.